### ISSUE TITLE
Why loss need use mean()

### ISSUE THREAD

**Issue by rmp918** [MadryLab/robustness#63]
Thank for this nice work 
I don't know why use mean() on loss 
https://github.com/MadryLab/robustness/blob/da6b6d31a13f37c7a2e97eb02c5f5551eb894a00/robustness/train.py#L449
Anyone can tell me why?

And I can't understand in the train.py which function call _model_loop
In this function we can call train and val , but val can use loss.backward()?
Hope someone can answer me,thanks~

**Comment by andrewilyas** [comment]
Hi, thanks for the questions:

1. ``.mean()`` is typically not needed but is useful if the user is supplying a custom_train_loss that returns a vector instead of a scalar.
2. The validation loop will always have gradients disabled unless you use the ``disable_no_grad`` option in ``train``, in which case you are allowed to use loss.backward() in the val loop.

Hope these answer your questions!
